src/System.Console/src/System/ConsolePal.Windows.cs

```diff
@@ -1207,7 +1207,7 @@ private static unsafe int WriteFileNative(IntPtr hFile, byte[] bytes, int offset
                     {
                         int numBytesWritten;
                         writeSuccess = (0 != Interop.Kernel32.WriteFile(hFile, p + offset, count, out numBytesWritten, IntPtr.Zero));
-                        Debug.Assert(!writeSuccess || count == numBytesWritten);
+                        Debug.Assert(writeSuccess);
```

Comment:
Previously we wouldn't assert if the WriteFile call returned false, but now this will?  If that's a change we want to make, what about the equivalent WriteConsole call below with a similar assert?

Reply:
Mm I'm inclined to delete this assert then while we figure out the right thing for both.

Reply:
Ok (and presumably the one below as well)

Reply:
I'll leave the other one, since it's a completely different API, and has not caused a problem.